packages/atomic/src/components/search/atomic-results-per-page/atomic-results-per-page.tsx

```diff
@@ -119,22 +120,27 @@ export class AtomicResultsPerPage implements InitializableComponent {
       return;
     }
 
+    const label = this.bindings.i18n.t('results-per-page');
+
     return (
       <div class="flex items-center">
         <span
           part="label"
           class="self-start text-on-background text-lg mr-3 leading-10"
+          aria-hidden="true"
         >
-          {this.bindings.i18n.t('results-per-page')}
+          {label}
         </span>
-        <div
-          part="buttons"
-          role="radiogroup"
-          aria-label={this.bindings.i18n.t('results-per-page')}
-          class="flex flex-wrap gap-2"
-        >
-          {this.choices.map((choice) => this.buildChoice(choice))}
-        </div>
+        <FieldsetGroup label={label}>
```

Comment:
Originally, I wanted to change the div into a `fieldset` and the `span` into `legend`, but the `legend` element has some weird default CSS which tailwind doesn't manage to completely remove on Safari.